meshapp/src/components/SignUp/SignUp.tsx

```diff
@@ -0,0 +1,225 @@
+import StepOne from "./StepOne";
+import StepTwo from "./StepTwo";
+import StepThree from "./StepThree";
+import Stepper from "@mui/material/Stepper";
+import Step from "@mui/material/Step";
+import StepLabel from "@mui/material/StepLabel";
+import { useState } from "react";
+import Grid from "@mui/material/Grid";
+import Button from "@mui/material/Button";
+import { useForm, SubmitHandler, FormProvider } from "react-hook-form";
+import ChevronLeftIcon from "@mui/icons-material/ChevronLeft";
+import ArrowForwardIcon from "@mui/icons-material/ArrowForward";
+import { Theme, ThemeProvider, Typography, createTheme } from "@mui/material";
+const steps = ["Create Account", "Verify Email", "Go to Profile"];
+
+//interface for the form field names and types
+export interface IFormInput {
+  firstName: string;
+  lastName: string;
+  nickName: string;
+  phoneNumber: string;
+  country: string;
+  state: string;
+  email: string;
+  password: string;
+  confirmPassword: string;
+  acceptedTermsConditions: boolean;
+  emailUpdates: boolean;
+  name: string;
+  location: string;
+  title: string;
+  label: string;
+  pronouns: string;
+  userType: string;
+  interests: Array<string>;
+  picture: File;
+}
+
+//styling for error
+const tooltipErrorTheme = createTheme({
+  components: {
+    MuiTooltip: {
+      styleOverrides: {
+        tooltip: {
+          backgroundColor: "#f44336",
+          color: "#ffffff",
+        },
+        arrow: {
+          color: "#f44336",
+        },
+      },
+    },
+  },
+});
+
+export default function SignUp() {
+  const formMethods = useForm<IFormInput>({
+    defaultValues: {
+      firstName: "",
+      lastName: "",
+      nickName: "",
+      phoneNumber: "",
+      country: "",
+      state: "",
+      email: "",
+      password: "",
+      confirmPassword: "",
+      acceptedTermsConditions: false,
+      emailUpdates: false,
+      name: "",
+      title: "",
+      label:
+        "This here could be your bio if you had one. Set one up as soon as you can to tell everyone about yourself. Bios help others learn about you.",
+      pronouns: "",
+      userType: "",
+      interests: [],
+    },
+    mode: "onBlur",
+  });
+
+  //contains which step we're on
+  const [activeStep, setActiveStep] = useState(0);
+
+  //goes to previous step
+  const handlePrevious = () => {
+    setActiveStep((prevActiveStep) => prevActiveStep - 1);
+  };
+
+  //handles the submission of the data in the current step
+  const handleContinue = async () => {
+    //function for continuing to next step
+    const continueToNext = () =>
+      setActiveStep((prevActiveStep) =>
+        prevActiveStep < steps.length ? prevActiveStep + 1 : prevActiveStep
+      );
+
+    let isValid = false;
+
+    // trigger validation for each step unless it is 2nd step (where there aren't any inputs)
+    //if we're on the last step, then trigger submit
+    if (activeStep === 0) {
```

Comment:
This is one place where having the steps as an enum would help. This could be `if (activeStep === Step.CREATE_ACCOUNT)` or something similar.

Reply:
So the activeStep variable is being used to both reference the index of the current form component in stepComponents and which label we are using in the stepper. Currently I just added in an enum, but I've tied each enum member to their corresponding index instead so that activeStep can still be used to both index the form component and be used to referenced to the enum. The conditionals for validation triggering now reflect what you suggested, but I'm still using the original step array (renamed to stepLabels) for the stepper to display.